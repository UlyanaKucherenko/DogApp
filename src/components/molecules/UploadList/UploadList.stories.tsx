import type { Meta, StoryObj } from '@storybook/react';

import { UploadList } from '.';

const meta = {
  title: 'Molecules/UploadList',
  component: UploadList,
  argTypes: {},
  parameters: {
    layout: 'fullscreen'
  }
} satisfies Meta<typeof UploadList>;

export default meta;
type Story = StoryObj<typeof meta>;

export const List: Story = {
  args: {
    data: [
      {
        breeds: [],
        id: '-Vo1XRVJ-',
        url: 'http://via.placeholder.com/640x360',
        width: 1399,
        height: 1050,
        sub_id: '',
        created_at: '2023-11-16T16:37:02.000Z',
        original_filename: 'test app.png',
        breed_ids: ''
      },
      {
        breeds: [],
        id: 'svssmNnjz',
        url: 'http://via.placeholder.com/640x360',
        width: 1399,
        height: 1050,
        sub_id: '',
        created_at: '2023-11-16T10:54:02.000Z',
        original_filename: 'test app.png',
        breed_ids: ''
      }
    ]
  }
};
