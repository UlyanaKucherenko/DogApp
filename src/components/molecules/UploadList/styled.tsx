import { Box } from '@mui/material';
import { styled } from '@mui/material/styles';
import theme from '../../../theme/theme';

const List = styled(Box)(() => ({
  display: 'grid',
  gridTemplateColumns: 'repeat(2, 1fr)',
  gridGap: '20px',

  [theme.breakpoints.down('sm')]: {
    gridTemplateColumns: 'repeat(1, 1fr)'
  }
}));

const ListItem = styled(Box)(() => ({
  width: '100%',
  height: '400px'
}));

export { List, ListItem };
