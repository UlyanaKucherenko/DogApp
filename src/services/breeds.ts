import { api } from './api';

export const breedsApi = api.injectEndpoints({
  endpoints: build => ({
    getBreeds: build.query<Breed[], void>({
      query: () => 'breeds'
    }),
    getBreedById: build.query<BreedDetails, string>({
      query: breedId => `breeds/${breedId}`
    })
  })
});

export const { useGetBreedsQuery, useGetBreedByIdQuery } = breedsApi;

export interface Breed {
  id: string;
  name: string;
  temperament: string;
}

export interface BreedDetails extends Breed {
  weight: {
    metric: string;
  };
  height: {
    metric: string;
  };
  lifeSpan: string;
  origin: string;
}
