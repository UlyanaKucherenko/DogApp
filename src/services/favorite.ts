import { api } from './api';

export interface Favorite {
  image_id: string;
  sub_id?: string;
}

export type FavoritesResponse = {
  id: number;
  user_id: string;
  image_id: string;
  sub_id: string;
  created_at: string;
  image: {
    id: string;
    url: string;
  };
};

export const favoritesApi = api.injectEndpoints({
  endpoints: build => ({
    addFavorite: build.mutation<unknown, Favorite>({
      query: body => ({
        url: `favourites`,
        method: 'POST',
        body
      }),
      invalidatesTags: [{ type: 'Favorites', id: 'LIST' }]
    }),

    deleteFavorite: build.mutation<unknown, { favourite_id: number }>({
      query: ({ favourite_id }) => ({
        url: `favourites/${favourite_id}`,
        method: 'DELETE'
      }),
      invalidatesTags: [{ type: 'Favorites', id: 'LIST' }]
    }),

    getFavorites: build.query<FavoritesResponse[], void>({
      query: () => ({
        url: `favourites`
      }),
      providesTags: (result = []) => [
        ...result.map(({ id }) => ({ type: 'Favorites', id }) as const),
        { type: 'Favorites' as const, id: 'LIST' }
      ]
    })
  })
});

export const {
  useAddFavoriteMutation,
  useDeleteFavoriteMutation,
  useGetFavoritesQuery
} = favoritesApi;
