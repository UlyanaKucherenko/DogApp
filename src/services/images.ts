import { api } from './api';

export interface Images {
  id: string;
  url: string;
  width: number;
  height: number;
  mime_type: string;
  breeds: [
    {
      id: number;
      name: string;
      weight: string;
      height: string;
      life_span: string;
      reference_image_id: string;
    }
  ];
  categories: [];
}

export interface UploadImage {
  breeds: [];
  id: string;
  url: string;
  width: number;
  height: number;
  sub_id: string;
  created_at: string;
  original_filename: string;
  breed_ids: string;
}

export type ImagesUploadResponse = {
  id: string;
  url: string;
  width: number;
  height: number;
  original_filename: string;
  pending: number;
  approved: number;
};

export interface BreedImage {
  id: string;
  url: string;
  reference_image_id: string;
}

type ImagesResponse = {
  [breedId: string]: BreedImage[];
};

export const imagesApi = api.injectEndpoints({
  endpoints: build => ({
    getBreedImages: build.query<ImagesResponse, string>({
      query: imageId => ({
        url: `images/${imageId}`
      }),
      providesTags: id => [{ type: 'Images', id }]
    }),

    getImagesId: build.query<
      ImagesResponse,
      {
        limit?: number;
        page?: number;
        mime_type?: string;
        order?: string;
        breed_id?: number;
      }
    >({
      query: ({
        limit = 10,
        page = 0,
        mime_type = 'all',
        breed_id,
        order = 'RANDOM'
      }) => ({
        url: `images/search?limit=${limit}&page=${page}&mime_types=${mime_type}&order=${order}&breed_id=${breed_id}`
      }),
      providesTags: (result = []) => [
        ...result.map(({ id }) => ({ type: 'Images', id }) as const),
        { type: 'Images' as const, id: 'LIST' }
      ]
    }),

    getImages: build.query<
      ImagesResponse,
      {
        limit?: number;
        page?: number;
        mime_type?: string;
        order?: string;
        breed_id?: number;
      }
    >({
      query: ({
        limit = 10,
        page = 0,
        mime_type = 'all',
        breed_id,
        order = 'RANDOM'
      }) => ({
        url: `images/search?limit=${limit}&page=${page}&mime_types=${mime_type}&order=${order}`
      }),
      providesTags: (result = []) => [
        ...result.map(({ id }) => ({ type: 'Images', id }) as const),
        { type: 'Images' as const, id: 'LIST' }
      ]
    }),

    uploadImage: build.mutation<Images, FormData>({
      query(formData) {
        return {
          url: `images/upload`,
          method: 'POST',
          body: formData
        };
      },
      invalidatesTags: [{ type: 'Images', id: 'LIST' }]
    }),
    uploadImagesList: build.query<
      UploadImage[],
      {
        limit?: number;
        page?: number;
      }
    >({
      query: ({ limit = 10, page = 0 }) => ({
        url: `images/?limit=${limit}&page=${page}`
      }),
      providesTags: (result = []) => [
        ...result.map(({ id }) => ({ type: 'Images', id }) as const),
        { type: 'Images' as const, id: 'LIST' }
      ]
    })
  })
});

export const {
  useGetBreedImagesQuery,
  useGetImagesIdQuery,
  useGetImagesQuery,
  useUploadImageMutation,
  useUploadImagesListQuery
} = imagesApi;
