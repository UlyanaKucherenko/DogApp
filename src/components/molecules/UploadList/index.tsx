import { UploadImage } from '../../../services/images';
import { GridCard } from '../GridCard';
import { List, ListItem } from './styled';

interface UploadListProps {
  data: UploadImage[];
}

export const UploadList = ({ data }: UploadListProps) => (
  <List>
    {(data || []).map(item => (
      <ListItem key={item.id}>
        <GridCard id={item.id} imgUrl={item.url} isFavorite cardType="upload" />
      </ListItem>
    ))}
  </List>
);
