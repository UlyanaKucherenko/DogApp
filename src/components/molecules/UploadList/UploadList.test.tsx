import { composeStories } from '@storybook/react';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import * as stories from './UploadList.stories';
import { store } from '../../../services/store';

describe('UploadList', () => {
  const { List } = composeStories(stories);
  test('should match snapshot UploadList', () => {
    const { container } = render(
      <Provider store={store}>
        <List />
      </Provider>
    );
    expect(container).toMatchSnapshot();
  });
});
