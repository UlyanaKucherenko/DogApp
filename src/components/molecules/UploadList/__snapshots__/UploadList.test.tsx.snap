// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UploadList should match snapshot UploadList 1`] = `
.emotion-0 {
  display: grid;
  grid-template-columns: repeat(2, 1fr);
  grid-gap: 20px;
}

@media (max-width:599.95px) {
  .emotion-0 {
    grid-template-columns: repeat(1, 1fr);
  }
}

.emotion-1 {
  width: 100%;
  height: 400px;
}

.emotion-2 {
  box-shadow: rgba(146, 31, 237, 1);
  border-radius: 20px;
  box-shadow: 6px 6px 0px 0px rgba(0, 0, 0, 1);
  -webkit-transition: box-shadow 0.3s ease-in-out;
  transition: box-shadow 0.3s ease-in-out;
  height: 100%;
}

.emotion-2:hover {
  box-shadow: 6px 6px 8px 0px rgba(25, 118, 210, 1);
}

.emotion-2:active {
  box-shadow: 6px 6px 8px 0px rgba(25, 118, 210, 1);
}

.emotion-3 {
  background-color: #fff;
  color: rgba(0, 0, 0, 0.87);
  -webkit-transition: box-shadow 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: box-shadow 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  border-radius: 4px;
  box-shadow: 0px 2px 1px -1px rgba(0,0,0,0.2),0px 1px 1px 0px rgba(0,0,0,0.14),0px 1px 3px 0px rgba(0,0,0,0.12);
  overflow: hidden;
  border-radius: 20px;
  box-shadow: none;
  position: relative;
  height: 100%;
}

.emotion-4 {
  display: block;
  -webkit-background-size: cover;
  background-size: cover;
  background-repeat: no-repeat;
  -webkit-background-position: center;
  background-position: center;
  width: 100%;
  object-fit: cover;
  height: 100%;
  object-fit: cover;
}

.emotion-5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 8px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  justify-content: space-between;
  padding: 24px 24px 20px 24px;
  position: absolute;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  width: 100%;
  height: 100%;
  background-color: transparent;
  -webkit-transition: background-color 0.3s ease-in-out;
  transition: background-color 0.3s ease-in-out;
}

.emotion-5>:not(:first-of-type) {
  margin-left: 8px;
}

.emotion-5:hover {
  background-color: rgba(0, 0, 0, 0.45);
}

.emotion-6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: end;
  -ms-flex-pack: end;
  -webkit-justify-content: flex-end;
  justify-content: flex-end;
  width: 100%;
}

<div>
  <div
    class="MuiBox-root emotion-0"
  >
    <div
      class="MuiBox-root emotion-1"
    >
      <div
        class="MuiBox-root emotion-2"
      >
        <div
          class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root emotion-3"
        >
          <img
            alt="item"
            class="MuiCardMedia-root MuiCardMedia-media MuiCardMedia-img emotion-4"
            src="http://via.placeholder.com/640x360"
          />
          <div
            class="MuiCardActions-root MuiCardActions-spacing emotion-5"
          >
            <div
              class="MuiBox-root emotion-6"
            >
              <div
                aria-label="add to favorites"
                class="MuiBox-root emotion-7"
              >
                <test-file-stub
                  aria-hidden="true"
                  classname="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                  focusable="false"
                />
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="MuiBox-root emotion-1"
    >
      <div
        class="MuiBox-root emotion-2"
      >
        <div
          class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root emotion-3"
        >
          <img
            alt="item"
            class="MuiCardMedia-root MuiCardMedia-media MuiCardMedia-img emotion-4"
            src="http://via.placeholder.com/640x360"
          />
          <div
            class="MuiCardActions-root MuiCardActions-spacing emotion-5"
          >
            <div
              class="MuiBox-root emotion-6"
            >
              <div
                aria-label="add to favorites"
                class="MuiBox-root emotion-7"
              >
                <test-file-stub
                  aria-hidden="true"
                  classname="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                  focusable="false"
                />
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
